dora/core/client/fs/src/main/java/alluxio/client/file/dora/WorkerLocationPolicy.java

```diff
@@ -1,265 +1,47 @@
-/*
- * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
- * (the "License"). You may not use this work except in compliance with the License, which is
- * available at www.apache.org/licenses/LICENSE-2.0
- *
- * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
- * either express or implied, as more fully set forth in the License.
- *
- * See the NOTICE file distributed with this work for information regarding copyright ownership.
- */
-
 package alluxio.client.file.dora;
 
-import static com.google.common.hash.Hashing.murmur3_32_fixed;
-import static java.lang.Math.ceil;
-import static java.lang.String.format;
-import static java.nio.charset.StandardCharsets.UTF_8;
-
-import alluxio.Constants;
 import alluxio.client.block.BlockWorkerInfo;
-import alluxio.wire.WorkerNetAddress;
+import alluxio.client.block.policy.BlockLocationPolicy;
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
 
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.Preconditions;
-import com.google.common.collect.ImmutableList;
-import com.google.common.hash.HashFunction;
-
-import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
-import java.util.NavigableMap;
-import java.util.Set;
-import java.util.TreeMap;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.AtomicReference;
-import java.util.concurrent.atomic.LongAdder;
-import java.util.stream.Collectors;
-import javax.annotation.Nullable;
-
-/**
- * An impl of WorkerLocationPolicy.
- */
-public class WorkerLocationPolicy {
-  private static final ConsistentHashProvider HASH_PROVIDER =
-      new ConsistentHashProvider(100, Constants.SECOND_MS);
-  private final int mNumVirtualNodes;
-
-  /**
-   * Constructs a new {@link WorkerLocationPolicy}.
-   *
-   * @param numVirtualNodes number of virtual nodes
-   */
-  public WorkerLocationPolicy(int numVirtualNodes) {
-    mNumVirtualNodes = numVirtualNodes;
-  }
 
+public interface WorkerLocationPolicy {
   /**
+   * TODO(jiacheng): the semantics here is ambiguous:
```

Comment:
see this @JiamingMai @dbw9580

Reply:
Right now seems we always want 1 and get 1 from here. But if we want to generalize this to an interface, some questions need answers, for example: For LocalWorkerPolicy, if there is no local worker, what should it return? If it just returns a random worker, then reader cannot find the cache written by a writer - the path just results to different random workers. 

That question actually implies that our policies need certain properties - for example it must be deterministic. If a writer resolves to a worker location, then all readers better resolve to that location too. 

Unfortunately, LocalWorkerPolicy is not perfectly deterministic. Local to writer does not mean local to readers. We should just treat this as an exception and probably only use it in tests.

I can't think of a way to FORCE all implementations of this interface to be determinstic. Strong javadoc comments is the strongest way I can imagine. I'm honestly concerned about opening this interface. It is the opposite of fool-proof!

Reply:
I think `LocalWorkerPolicy` may return an empty list. If there is a `LocalWorkerPerferredPolicy`, then I think it should return a list that includes local node. Moreover, if there isn't any local node, I will expect `LocalWorkerPerferredPolicy` to return a list with remote nodes.

Referring to the practice of the industry, it seems to be a common practice to clarify its behavior through comments.